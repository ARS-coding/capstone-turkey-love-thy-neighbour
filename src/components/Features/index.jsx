import React from "react";

import { Container, Row, Col, Card } from "react-bootstrap";

import { ReactComponent as NetworkImg } from "../../images/network.svg";
import { ReactComponent as SpeechBubbleImg } from "../../images/speech-bubble.svg";
import { ReactComponent as CouponImg } from "../../images/coupon.svg";

import "./index.scss";

// TODO: add real content

function Features() {
  return (
    <section className="features-section container-fluid">
      <Container className="features-content-container d-flex flex-column justify-content-center">
        <h2>Features</h2>
        <Row className="features-content-row d-md-flex justify-content-md-around text-center">
          <Col xs={12} sm={12} md={3}>
            <Card className="d-flex flex-column align-items-center justify-content-center">
              <NetworkImg className="network-svg img-fluid" />
              <Card.Title>Meet</Card.Title>
              <Card.Text>
                Some quick example text to build on the card title and make up
                the bulk of the card&lsquo;s content.
              </Card.Text>
            </Card>
          </Col>
          <Col xs={12} sm={12} md={3} className="my-1 my-sm-2 my-md-0">
            <Card className="d-flex flex-column align-items-center justify-content-center">
              <SpeechBubbleImg className="speech-bubble-svg img-fluid" />

              <Card.Title>Card Title</Card.Title>
              <Card.Text>
                Some quick example text to build on the card title and make up
                the bulk of the card&lsquo;s content.
              </Card.Text>
            </Card>
          </Col>
          <Col xs={12} sm={12} md={3}>
            <Card className="d-flex flex-column align-items-center justify-content-center">
              <CouponImg className="coupon-svg img-fluid" />

              <Card.Title>Coupons</Card.Title>
              <Card.Text>
                Some quick example text to build on the card title and make up
                the bulk of the card&lsquo;s content.
              </Card.Text>
            </Card>
          </Col>
        </Row>
      </Container>
    </section>
  );
}

export default Features;
