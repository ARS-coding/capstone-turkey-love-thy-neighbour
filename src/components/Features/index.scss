@import "../../style/variables";

.features-section {
  .features-content-container {
    h2 {
      color: $logo-red;
      text-align: center;
      @media screen and (min-width: 768px) {
        margin-bottom: 30px;
        font-size: 40px;
      }
    }

    .features-content-row {
      margin: 0 !important;

      & > div {
        // columns inside the row
        padding: 0;

        div.card {
          // card divs inside the columns
          margin: 0;
          padding: 10px;
          transition: all ease-in-out 300ms;
          box-shadow: rgba(60, 64, 67, 0.3) 0px 1px 2px 0px,
            rgba(60, 64, 67, 0.15) 0px 2px 6px 2px;

          @media screen and (min-width: 342px) {
            height: 153px;
          }
          @media screen and (min-width: 576px) {
            width: 300px;
            height: 170px;
          }
          @media screen and (max-width: 767px) {
            width: 100% !important;
          }
          @media screen and (min-width: 768px) {
            margin: 0 auto;
            height: 245px;
            width: 100% !important;
          }

          & > svg {
            fill: $logo-blue;
            transition: all ease-in-out 300ms;

            @media screen and (min-width: 362px) {
              min-width: 70px;
              max-width: 120px;
              min-height: 35px;
              max-height: 70px;
            }

            @media screen and (max-width: 361px) {
              height: 59px !important;
              width: 70px !important;
            }

            &.network-svg {
              margin-bottom: 7px;
            }

            &.speech-bubble-svg {
              margin-bottom: 3px;
            }
          }

          &:hover {
            // whenever the card gets hovered on...
            background-color: #8dd6f5;
            transition: all ease-in-out 300ms;

            svg {
              transition: all ease-in-out 300ms;
              fill: $black;
            }

            .card-title {
              color: $logo-red;
              transition: all ease-in-out 300ms;
            }

            .card-text {
              color: $white;
              transition: all ease-in-out 300ms;
            }
          }

          & > .card-title {
            color: $logo-orange;
            transition: all ease-in-out 300ms;

            @media screen and (min-width: 992px) and (max-width: 1259px) {
              font-size: 25px;
            }
            @media screen and (min-width: 1260px) {
              font-size: 30px;
            }
          }

          & > p.card-text {
            transition: all ease-in-out 300ms;

            @media screen and (max-width: 341px) {
              font-size: 11px;
            }
            @media screen and (min-width: 342px) and (max-width: 1259px) {
              font-size: 13px;
            }
            @media screen and (min-width: 1260px) {
              font-size: 15px;
            }
          }
        }
      }
    }
  }
}
