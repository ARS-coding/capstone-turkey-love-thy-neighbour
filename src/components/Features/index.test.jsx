import React from "react";
import renderer from "react-test-renderer";
import Features from ".";

describe("Features Section", () =>
  it("renders Features section correctly", () => {
    const tree = renderer.create(<Features />).toJSON();
    expect(tree).toMatchSnapshot();
  }));
