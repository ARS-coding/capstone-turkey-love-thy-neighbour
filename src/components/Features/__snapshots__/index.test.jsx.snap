// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Features Section renders Features section correctly 1`] = `
<section
  className="features-section container-fluid"
>
  <div
    className="features-content-container d-flex flex-column justify-content-center container"
  >
    <h2>
      Features
    </h2>
    <div
      className="features-content-row d-md-flex justify-content-md-around text-center row"
    >
      <div
        className="col-md-3 col-sm-12 col-12"
      >
        <div
          className="d-flex flex-column align-items-center justify-content-center card"
        >
          <svg
            className="network-svg img-fluid"
          >
            network.svg
          </svg>
          <div
            className="card-title h5"
          >
            Meet
          </div>
          <p
            className="card-text"
          >
            Find people in your neighborhood and expand your network.
          </p>
        </div>
      </div>
      <div
        className="my-1 my-sm-2 my-md-0 col-md-3 col-sm-12 col-12"
      >
        <div
          className="d-flex flex-column align-items-center justify-content-center card"
        >
          <svg
            className="speech-bubble-svg img-fluid"
          >
            speech-bubble.svg
          </svg>
          <div
            className="card-title h5"
          >
            Filtering
          </div>
          <p
            className="card-text"
          >
            Filter nearby people by their age, gender and interests.
          </p>
        </div>
      </div>
      <div
        className="col-md-3 col-sm-12 col-12"
      >
        <div
          className="d-flex flex-column align-items-center justify-content-center card"
        >
          <svg
            className="coupon-svg img-fluid"
          >
            coupon.svg
          </svg>
          <div
            className="card-title h5"
          >
            Coupons
          </div>
          <p
            className="card-text"
          >
            Get coupons from our sponsored places and pay less.
          </p>
        </div>
      </div>
    </div>
  </div>
</section>
`;
